src/python/pants/backend/native/register.py

```diff
@@ -48,6 +48,7 @@ def register_goals():
   # TODO(#5962): register these under the 'compile' goal when we eliminate the product transitive
   # dependency from export -> compile.
   task(name='conan-prep', action=ConanPrep).install('native-compile')
+  # TODO: deprecate the `native-third-party-fetch` task name in favor of `conan-fetch` somehow!
```

Comment:
This is possible by putting a deprecated scope on the task, but fine to do in a followup.

Reply:
When I was testing `deprecated_scope` I had forgotten that pantsd was enabled, and this led to an options error which made me think there was a bug in `deprecated_scope`, which I didn't write down here. This is working when I test it now, so I will add the deprecation.